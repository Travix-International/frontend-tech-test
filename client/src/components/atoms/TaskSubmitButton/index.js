import React from 'react'
import PropTypes from 'prop-types'
import './index.css'

const TaskSubmitButton = ({ disabled, onClick }) => {
  const disabledProp = disabled ? 'true' : null
  return (
    <button
      className='task-submit-button'
      disabled={disabledProp}
      onClick={onClick}
    >
      Save
    </button>
  )
}

TaskSubmitButton.propTypes = {
  disabled: PropTypes.bool.isRequired,
  onClick: PropTypes.func
}

export default TaskSubmitButton
