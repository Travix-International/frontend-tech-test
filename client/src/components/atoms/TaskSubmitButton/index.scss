@import 'src/styles/base/colors.scss';
@import 'src/styles/base/button.scss';

.task-submit-button {
  @extend .button;

  color: $green;

  &:hover {
    color: darken($green, 10%);
  }

  &:active {
    color: darken($green, 20%);
  }

  &:disabled {
    color: lighten($black, 50%);
    cursor: not-allowed;
  }
}
