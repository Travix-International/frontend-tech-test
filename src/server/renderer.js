import React from 'react';
import ReactServer from 'react-dom/server';
import { StaticRouter as Router, matchPath } from 'react-router';
import { Provider } from 'react-redux';
import { Helmet } from 'react-helmet';
import { /* get as _get, */ isFunction as _isFunction } from 'lodash';
import renderFullPage from './layout';
import configureStore from '../configureStore';
import { entryRouteComponentMap, getEntryAndRoute, entryReducerMap } from './entryAndRoute';

function applyInitStore(req) {
  const entryRouteInfo = getEntryAndRoute(req.path);
  const currentEntry = entryRouteInfo.entry;
  const rootReducer = entryReducerMap[currentEntry];
  const store = configureStore({}, rootReducer);
  req.reduxStore = store;
  req.entryRouteInfo = entryRouteInfo;
  // next();
}

function applyRouteCheckResult(req) {
  const { path, reduxStore: store, query, entryRouteInfo } = req;
  const promises = [];
  const currentEntry = entryRouteInfo.entry;
  const route = entryRouteInfo.route;
  const match = matchPath(path, route);
  const { loadData } = route;

  // set currentEntry to req
  req.currentEntry = currentEntry;

  // it's possible no loadData set
  if (_isFunction(loadData)) {
    let action = loadData(match, query);
    if (action) {
      if (_isFunction(action) || action.type) {
        action = store.dispatch(action);
        if (action.then) {
          promises.push(action);
        }
      }
    }
  }

  return Promise.all(promises);
}

function responsePage(req, res, clientStats) {
  const { url, reduxStore: store, currentEntry } = req;
  const routerContext = {};
  const reduxStateString = JSON.stringify(store.getState()).replace(/</g, '\\u003c');

  const RouteComponent = entryRouteComponentMap[currentEntry];

  const content = (
    <Provider store={store}>
      <Router context={routerContext} location={url}>
        <RouteComponent />
      </Router>
    </Provider>
  );

  // render to sting to get helmet setting
  const contentString = ReactServer.renderToString(content);
  const head = Helmet.renderStatic();
  const htmlString = renderFullPage(contentString, reduxStateString, head, currentEntry, clientStats);

  res.send(htmlString);
}

export default function renderer({ clientStats }) {
  return (req, res) => {
    applyInitStore(req);
    return applyRouteCheckResult(req).then(() => {
      return responsePage(req, res, clientStats);
    });
  };
}
