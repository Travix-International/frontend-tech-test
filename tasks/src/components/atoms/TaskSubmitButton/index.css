.button, .task-submit-button {
  align-items: center;
  appearance: none;
  background-color: transparent;
  border: 0;
  cursor: pointer;
  display: flex;
  font-size: 13px;
  font-weight: 500;
  height: 50px;
  justify-content: center;
  line-height: 1.3;
  overflow: hidden;
  padding: 12px;
  text-align: center;
  text-decoration: none;
  text-transform: uppercase;
  width: 50px; }
  .button:hover, .task-submit-button:hover {
    transition: all 60ms ease; }
  .button:active, .task-submit-button:active {
    transition: all 60ms ease; }

.task-submit-button {
  color: #090; }
  .task-submit-button:hover {
    color: #006600; }
  .task-submit-button:active {
    color: #003300; }
