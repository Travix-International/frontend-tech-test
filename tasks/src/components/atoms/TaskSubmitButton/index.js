import React from 'react'
import PropTypes from 'prop-types'

const TaskSubmitButton = ({ onClick }) => {
  return (
    <button onClick={onClick}>
      Submit
    </button>
  )
}

TaskSubmitButton.propTypes = {
  onClick: PropTypes.func
}

export default TaskSubmitButton
