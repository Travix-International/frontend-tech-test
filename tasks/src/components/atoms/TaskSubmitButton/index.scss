@import 'src/styles/base/colors.scss';
@import 'src/styles/base/button.scss';

.task-submit-button {
  @extend .button;

  color: $green;

  &:hover {
    color: darken($green, 10%);
  }

  &:active {
    color: darken($green, 20%);
  }
}
