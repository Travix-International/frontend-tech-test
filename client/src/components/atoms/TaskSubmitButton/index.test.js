import React from 'react'
import renderer from 'react-test-renderer'
import TaskSubmitButton from './index'

describe('TaskSubmitButton', () => {
  it('should render correctly', () => {
    const onClick = jest.fn()
    const tree = renderer.create(
      <TaskSubmitButton onClick={onClick} disabled={false} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render correctly (disabled button)', () => {
    const onClick = jest.fn()
    const tree = renderer.create(
      <TaskSubmitButton onClick={onClick} disabled={true} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
