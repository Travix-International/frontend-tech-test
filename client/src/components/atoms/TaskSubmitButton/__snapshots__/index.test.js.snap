// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskSubmitButton should render correctly (disabled button) 1`] = `
<button
  className="task-submit-button"
  disabled="true"
  onClick={[Function]}
>
  Save
</button>
`;

exports[`TaskSubmitButton should render correctly 1`] = `
<button
  className="task-submit-button"
  disabled={null}
  onClick={[Function]}
>
  Save
</button>
`;
